web-ui/src/main/resources/catalog/components/viewer/ViewerModule.js

```diff
@@ -245,24 +252,104 @@
       // watch service data: when a profile graph is available, render it
       var me = this;
       $scope.$watch(
-          function() {
-            return gnViewerService.getProfileGraphData();
-          },
-          function(newData, oldData) {
-            me.profileGraph = newData && JSON.parse(newData).profile;
-          }
+        function() {
+          return gnViewerService.getProfileGraphData();
+        },
+        function(newData, oldData) {
+          me.profileGraph = newData && JSON.parse(newData).profile;
+        }
       );
+
+      // this is used to render hovered point on the profile
+      this.hoveredProfilePoint = new ol.Feature();
+      var source = new ol.source.Vector();
+      var hoveredPointLayer = new ol.layer.Vector({
+        source: source
+      });
+      source.addFeature(this.hoveredProfilePoint);
+      hoveredPointLayer.setZIndex(1000);
+      map.addLayer(hoveredPointLayer);
+
+      // let's get the geometry tool layer (where profiles are drawn)
+      var profileVectorLayer = gnGeometryService.getCommonLayer(map);
+
+      // show height on graph when hovering the feature
+      this.profileHighlight = -1;
+      map.on('pointermove', function (evt) {
```

Comment:
Put all this code in a specific file (service, directive or class).
This is specific and polluates the `viewerModule`.